libs/community/langchain_community/document_loaders/sharepoint.py

```diff
@@ -52,7 +64,12 @@ def lazy_load(self) -> Iterator[Document]:
             if not isinstance(target_folder, Folder):
                 raise ValueError(f"There isn't a folder with path {self.folder_path}.")
             for blob in self._load_from_folder(target_folder):
-                yield from blob_parser.lazy_parse(blob)
+                for parsed_blob in blob_parser.lazy_parse(blob):
+                    auth_identities = self.authorized_identities(
+                        self.document_library_id, self.file_id
+                    )
+                    parsed_blob.metadata["authorized_identities"] = auth_identities
+                    yield parsed_blob
         if self.folder_id:
             target_folder = drive.get_item(self.folder_id)
             if not isinstance(target_folder, Folder):
```

Comment:
within lazy_load method, it seems all the yield_from would require same opration as this.